### PR TITLE
[NativeAOT/x86] Remove unused assembly files

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/i386/ThunkPoolThunks.asm
Comment (am11) [inline src/coreclr/nativeaot/Runtime/i386/ThunkPoolThunks.asm:1]: Is it also unused on other archs?

Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/i386/ThunkPoolThunks.asm:1]: No, it's used on macOS and other Apple platforms.

**Comment by jkotas** [comment]
/ba-g Infrastructure timeouts. The affected legs do not exercise the change in meaningful way, no reason for retrying.

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**